From a307fa54968d88bbccf6c9b37e7a394edcfaf20e Mon Sep 17 00:00:00 2001
From: Hou Zhiqiang <Zhiqiang.Hou@nxp.com>
Date: Thu, 3 Nov 2022 16:49:43 +0800
Subject: [PATCH 40/64] configs: imx8mm: Move back the SGI mailbox and RPMSG
 memory entries definitions

Move back the definitions of SGI mailbox and RPMSG regions, so that keep
the IVSHMEM regions on the top.

Signed-off-by: Hou Zhiqiang <Zhiqiang.Hou@nxp.com>
---
 configs/arm64/imx8mm-harpoon-rtos.h | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/configs/arm64/imx8mm-harpoon-rtos.h b/configs/arm64/imx8mm-harpoon-rtos.h
index 1890de63..3f50860a 100644
--- a/configs/arm64/imx8mm-harpoon-rtos.h
+++ b/configs/arm64/imx8mm-harpoon-rtos.h
@@ -32,15 +32,15 @@ typedef enum IRQn {
 } IRQn_Type;
 
 #define RTOS_COMMON_MEMORY_REGIONS \
-	MMIO_REGION_RWS(0xb8500000, 0xb8500000, KB(4)),  /* SGI MBOX */             \
-	MMIO_REGION_RWS(0xb8600000, 0xb8600000, KB(64)), /* RPMSG */                \
-	MEM_REGION_RWS( 0xb8700000, 0xb8700000, MB(1)),  /* RPMSG BUF */            \
 	MEM_REGION_ROS( 0xbbaf0000, 0xbbaf0000, KB(4)),  /* IVSHMEM */              \
 	MEM_REGION_RWS( 0xbbaf1000, 0xbbaf1000, KB(36)), /* IVSHMEM */              \
 	MEM_REGION_ROS( 0xbbafa000, 0xbbafa000, KB(8)),  /* IVSHMEM */              \
 	MEM_REGION_RWS( 0xbbafc000, 0xbbafc000, KB(8)),  /* IVSHMEM */              \
 	MEM_REGION_ROS( 0xbbafe000, 0xbbafe000, KB(8)),  /* IVSHMEM */              \
 	/* IVSHMEM regions should always be kept on top so PCI_IVS doesn't change */\
+	MMIO_REGION_RWS(0xb8500000, 0xb8500000, KB(4)),  /* SGI MBOX */             \
+	MMIO_REGION_RWS(0xb8600000, 0xb8600000, KB(64)), /* RPMSG */                \
+	MEM_REGION_RWS( 0xb8700000, 0xb8700000, MB(1)),  /* RPMSG BUF */            \
 	MMIO_REGION_RW( 0x302d0000, 0x302d0000, KB(64)), /* GPT1 */                 \
 	MMIO_REGION_RW( 0x302e0000, 0x302e0000, KB(64)), /* GPT2 */                 \
 	MMIO_REGION_RWS(0x30330000, 0x30330000, KB(64)), /* IOMUXC */               \
-- 
2.34.1

